server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcResolver.java

```diff
@@ -129,110 +162,88 @@
-        int column_index = 0;
+        int columnIndex = 0;
+
         for (OneField oneField : record) {
-            ColumnDescriptor column = columns.get(column_index);
-            if (LOG.isWarnEnabled() && DataType.get(column.columnTypeCode()) != DataType.get(oneField.type)) {
-                LOG.warn("The provided tuple of data may be disordered. Datatype of column with descriptor '{}' must be '{}', but actual is '{}'",
-                        column,
-                        DataType.get(column.columnTypeCode()),
-                        DataType.get(oneField.type));
-            }
+            ColumnDescriptor column = columns.get(columnIndex++);
 
-            // Check that data type is supported
-            switch (DataType.get(oneField.type)) {
-                case BOOLEAN:
-                case INTEGER:
-                case FLOAT8:
-                case REAL:
-                case BIGINT:
-                case SMALLINT:
-                case NUMERIC:
-                case VARCHAR:
-                case BPCHAR:
-                case TEXT:
-                case BYTEA:
-                case TIMESTAMP:
-                case DATE:
-                    break;
-                default:
-                    throw new UnsupportedOperationException(
-                            String.format("Field type '%s' (column '%s') is not supported",
-                                    DataType.get(oneField.type),
-                                    column));
-            }
+            DataType oneFieldType = DataType.get(oneField.type);
+            DataType columnType = DataType.get(column.columnTypeCode());
 
-            if (LOG.isDebugEnabled() &&
-                    DataType.get(oneField.type) == DataType.BYTEA) {
-                String converted = (oneField.val != null) ? new String((byte[]) oneField.val) : "null";
-                LOG.debug("OneField content (conversion from BYTEA): '{}'", converted);
+            if (!DATATYPES_SUPPORTED.contains(oneFieldType)) {
+                throw new UnsupportedOperationException(
+                        String.format("Field type '%s' (column '%s') is not supported",
+                                oneFieldType, column));
             }
 
-            // Convert TEXT columns into native data types
-            if ((DataType.get(oneField.type) == DataType.TEXT) && (DataType.get(column.columnTypeCode()) != DataType.TEXT)) {
-                oneField.type = column.columnTypeCode();
-                if (oneField.val != null) {
-                    String rawVal = (String) oneField.val;
-
-                    LOG.debug("OneField content (conversion from TEXT): '{}'", rawVal);
-                    switch (DataType.get(column.columnTypeCode())) {
-                        case VARCHAR:
-                        case BPCHAR:
-                        case TEXT:
+            if (LOG.isDebugEnabled()) {
+                if (oneField.val == null) {

```

Comment:
Maybe something like this:
```suggestion
                String converted;
                if (oneField.val == null) {
                    converted = "null";
                } else if (oneFieldType == DataType.BYTEA) {
                    converted = "'" + new String((byte[]) oneField.val) + "'";
                } else {
                    converted = "'" + oneField.val.toString() + "'";
                }
                LOG.debug("Column {} OneField: type {}, content {}", columnIndex, oneFieldType, converted);
```